### ISSUE TITLE
Typing changes required for original type hinting

### ISSUE THREAD

**Issue by goldingdamien** [origin-1/gulp-eslint-new#8]
It seems the following is required to get type hinting working as originally possible in gulp-eslint:

```js
// const eslint = require('gulp-eslint')
/**
 * @type {import('gulp-eslint-new')['default']}
 */
const eslint = require('gulp-eslint-new')
```

**Comment by fasttime** [comment]
Thanks! Do you know how to fix this?

**Comment by goldingdamien** [comment]
The typings file(https://github.com/origin-1/gulp-eslint-new/blob/main/lib/gulp-eslint-new.d.ts) seems to be exporting a default type while the actual code(https://github.com/origin-1/gulp-eslint-new/blob/main/lib/gulp-eslint-new.js) is using module.exports which I believe is handled differently to an es6 default export.

I am not completely sure, but I think declaring a module is probably the way it is done:
https://stackoverflow.com/a/41707197/1764521

gulp-eslint doesn't seem to be using any typings: https://github.com/adametry/gulp-eslint

**Comment by fasttime** [comment]
I checked other [`@types/*`](https://www.npmjs.com/~types) packages, and it seems that CommonJS modules with a default export are all typed with the `export =` syntax. TypeScript itself doesn't have a problem with the `export default` syntax if option `esModuleInterop` is set to `true`, but probably IDEs get confused by the combination of `require` with a module declaration.

So I went ahead and drafted [a fix](https://github.com/origin-1/gulp-eslint-new/pull/9) that switches the type declaration file to use the `export =` syntax. Now type hints are working again in VSCode. I'd like to test this change in other IDEs when I have a chance.
